### DISCUSSION TITLE
"partials/content/figure.html" at <index .Params.images 0>: error calling index: index of untyped nil

### DISCUSSION THREAD

**Discussion by papanito** [thuliteio/doks#984] [Q&A]
When running `hugo` I get this error:

```
ERROR 2023/01/14 19:05:11 render of "page" failed: "/home/aedu/Workspaces/papanito/web.papanito/layouts/blog/single.html:29:9": execute of template failed: template: blog/single.html:29:9: executing "main" at <partial "content/figure.html" .>: error calling partial: "/home/aedu/Workspaces/papanito/web.papanito/layouts/partials/content/figure.html:6:49": execute of template failed: template: partials/content/figure.html:6:49: executing "partials/content/figure.html" at <index .Params.images 0>: error calling index: index of untyped nil
Error: Error building site: failed to render pages: render of "page" failed: "/home/aedu/Workspaces/papanito/web.papanito/layouts/blog/single.html:29:9": execute of template failed: template: blog/single.html:29:9: executing "main" at <partial "content/figure.html" .>: error calling partial: "/home/aedu/Workspaces/papanito/web.papanito/layouts/partials/content/figure.html:6:49": execute of template failed: template: partials/content/figure.html:6:49: executing "partials/content/figure.html" at <index .Params.images 0>: error calling index: index of untyped nil
```

The line `figure.html.6.49` looks like this

```
{{ $image := .Resources.GetMatch (printf "**%s" (index .Params.images 0)) }}
```

My  `config/_default/params.toml` contains the `images` parameter

```
images = ["logo_w_blue.png"]
```

The file `logo_w_blue.png` is in folder `static`. 

So not sure why it's not building anymore.

**Comment by papanito** [comment]
What puzzles me most, I can build it successfully locally, but it fails running it in the ci pipeline

**Reply by iakat** [reply]
I have the same error. Have you figured it out?

**Reply by papanito** [reply]
unfortunately not, I just removed the failing code from the template, so the page is built again

**Comment by h-enk** [comment]
Note that, the `images` definition in `./config/_default/params.toml` is for Open Graph tags only:

```toml
[..]
## Open Graph
images = ["doks.png"]
[..]
```

And, gets rendered with `./layouts/partials/head/opengraph.html`. So, not with `./layouts/partials/content/figure.html`.

I think it's more likely that you have somewhere in the frontmatter of one of your content files `images: [""]` in stead of `images: []`. The upcoming new Doks 1.0 will handle this more smartly, preventing this issue to occur.

If you need further assistance with this, please share a repo

**Reply by papanito** [reply]
I will have a look. Interestingly still is that local build is actually fine.

**Reply by papanito** [reply]
I replaced all findings of `images: [""]` with `images: []` (I had 3 occurrences), however the build still fails :-(

I added you to my repo as Developer in Gitlab.

**Comment by h-enk** [comment]
I noticed a few things.

1. Running `npm run start` got me a warning: `WARNING: Found no color leading to 4.5:1 contrast ratio against #4f81bd...` Changed (not necessary) [_varaiables.scss](https://gitlab.com/wyssmann/website/-/blob/master/assets/scss/common/_variables.scss#L28):

```scss
[..]
/* basic color scheme */
// $primary: #4f81bd;
$primary: $purple;
[..]
```

3. [What is as a service mesh?](https://gitlab.com/wyssmann/website/-/blob/master/content/en/blog/2021-05-what-is-a-service-mesh/index.md?plain=1#L22) contains a remote image (not supported by `@hyas/images` 0.2.2). Commented this out:

```md
[..]
{{</* img src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/51/Overview_of_a_three-tier_application_vectorVersion.svg/512px-Overview_of_a_three-tier_application_vectorVersion.svg.png" caption="Bartledan (talk), based on a file by User:Foofy, Public domain, via Wikimedia Commons" */>}}
[..]
```

4. [Do you spot it ...](https://gitlab.com/wyssmann/website/-/blob/master/content/en/blog/2019-12-do-you-spot-it/index.md?plain=1#L2) contains three dots in the title (Hugo does not like that :wink:). Changed that to:

```md
---
title: Do you spot it?
[..]
```

5. [`postcss.config.js`](https://gitlab.com/wyssmann/website/-/blob/master/config/postcss.config.js#L13-14) is missing some `safelist classes`. Added:

```js
[..]
      safelist: [
        'lazyloaded',
        'table',
        'thead',
        'tbody',
        'tr',
        'th',
        'td',
        'h5',
        'alert-link',
        'container-xxl',
        'container-fluid',
        'offcanvas-backdrop',
        'img-fluid',
        'lazyload',
        'blur-up',
        'figcaption',
[..]

```

Now, running `npm run build` succeeds successfully and show like intended in production:

![Snag_3614d9](https://user-images.githubusercontent.com/3902872/228154936-c40c2015-05d5-4644-8b79-3b885af4f560.png)
